apps/wallet-dashboard/components/Dialogs/migration/MigrationDialog.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState } from 'react';
+import { useCurrentAccount, useSignAndExecuteTransaction } from '@iota/dapp-kit';
+import { IotaObjectData } from '@iota/iota-sdk/client';
+import { useMigrationTransaction } from '@/hooks/useMigrationTransaction';
+import { Dialog } from '@iota/apps-ui-kit';
+import toast from 'react-hot-toast';
+import { TransactionDialogView } from '../TransactionDialog';
+import { MigrationDialogView } from './enums';
+import { ConfirmMigrationView } from './views';
+
+interface MigrationDialogProps {
+    handleClose: () => void;
+    basicOutputObjects: IotaObjectData[] | undefined;
+    nftOutputObjects: IotaObjectData[] | undefined;
+    onSuccess: (digest: string) => void;
+    setOpen: (bool: boolean) => void;
+    open: boolean;
+    isTimelocked: boolean;
+}
+
+export function MigrationDialog({
+    handleClose,
+    basicOutputObjects = [],
+    nftOutputObjects = [],
+    onSuccess,
+    open,
+    setOpen,
+    isTimelocked,
+}: MigrationDialogProps): JSX.Element {
+    const account = useCurrentAccount();
+    const [txDigest, setTxDigest] = useState<string>('');
+    const [view, setView] = useState<MigrationDialogView>(MigrationDialogView.Confirmation);
+
+    const {
+        data: migrateData,
+        isPending: isMigrationPending,
+        isError: isMigrationError,
+    } = useMigrationTransaction(account?.address || '', basicOutputObjects, nftOutputObjects);
+
+    const { mutateAsync: signAndExecuteTransaction, isPending: isSendingTransaction } =
+        useSignAndExecuteTransaction();
+
+    async function handleMigrate(): Promise<void> {
+        if (!migrateData) return;
+        signAndExecuteTransaction(
+            {
+                transaction: migrateData.transaction,
+            },
+            {
+                onSuccess: (tx) => {
+                    onSuccess(tx.digest);
```

Comment:
I don't think it makes much sense to use a useMutation because this is a very specific case, here we are using `onSuccess`, `setTxDigest` & `setView`, sometimes its better to duplicate a bit more of code rather than having a bad abstraction. Your case was different because you are doing a custom load+error system, but we are not doing that here

SO I think this is fine

But thanks for keeping the comment from the other day in mind!